src/spikeinterface/widgets/amplitudes.py

```diff
@@ -73,13 +82,11 @@ def __init__(
         amplitudes_segment = amplitudes[segment_index]
         total_duration = sorting_analyzer.get_num_samples(segment_index) / sorting_analyzer.sampling_frequency
 
-        spiketrains_segment = {}
-        for i, unit_id in enumerate(sorting.unit_ids):
-            times = sorting.get_unit_spike_train(unit_id, segment_index=segment_index)
-            times = times / sorting.get_sampling_frequency()
-            spiketrains_segment[unit_id] = times
+        all_spiketrains = {
+            unit_id: sorting.get_unit_spike_train(unit_id, segment_index=0, return_times=True)
```

Comment:
```suggestion
            unit_id: sorting.get_unit_spike_train(unit_id, segment_index=segment_index, return_times=True)
```
this only defaults to 0 if the `sorting_index=None` so in the case the user gives an index we should use it.

Reply:
Great, this was propogated forward, so I think this counts as you adding a new feature @zm711 !